packages/vite/src/node/plugins/css.ts

```diff
@@ -818,6 +784,79 @@ export function cssPostPlugin(config: ResolvedConfig): Plugin {
   }
 }
 
+/**
+ * Create a replacer function that takes code and replaces given pure CSS chunk imports
+ * @param pureCssChunkNames The chunks that only contain pure CSS and should be replaced
+ * @param outputFormat The module output format to decide whether to replace `import` or `require`
+ */
+export function getEmptyChunkReplacer(
+  pureCssChunkNames: string[],
+  outputFormat: ModuleFormat,
+): (code: string) => string {
+  const emptyChunkFiles = pureCssChunkNames
+    .map((file) => path.basename(file))
+    .join('|')
+    .replace(/\./g, '\\.')
+
+  // require and import calls might be chained by minifier using the comma operator
+  // in this case we have to keep one comma
+  // if a next require is chained or add a semicolon to terminate the chain.
+  const emptyChunkRE = new RegExp(
+    outputFormat === 'es'
+      ? `\\bimport\\s*["'][^"']*(?:${emptyChunkFiles})["'];\n?`
+      : `\\brequire\\(\\s*["'][^"']*(?:${emptyChunkFiles})["']\\);\n?`,
+    'g',
+  )
+
+  return (code: string) =>
+    code.replace(
+      emptyChunkRE,
+      // remove css import while preserving source map location
+      (m) => `/* empty css ${''.padEnd(m.length - 15)}*/`,
+    )
+}
+
+/**
+ * Remove pure CSS chunks from the output bundle
+ * @param bundle The output bundle
+ * @param pureCssChunkNames Array of pure CSS chunk names
+ * @param outputFormat The current output format, to decide whether `require` or `import` is used
+ */
+export function removePureCssChunks(
+  bundle: OutputBundle,
+  pureCssChunkNames: string[],
+  outputFormat: ModuleFormat,
+): void {
+  const replaceEmptyChunk = getEmptyChunkReplacer(
+    pureCssChunkNames,
+    outputFormat,
+  )
+
+  for (const file in bundle) {
+    const chunk = bundle[file]
+    if (chunk.type === 'chunk') {
+      // remove pure css chunk from other chunk's imports,
+      // and also register the emitted CSS files under the importer
+      // chunks instead.
+      chunk.imports = chunk.imports.filter((file) => {
+        if (pureCssChunkNames.includes(file)) {
+          const { importedCss, importedAssets } = (bundle[file] as OutputChunk)
+            .viteMetadata!
+          importedCss.forEach((file) =>
+            chunk.viteMetadata!.importedCss.add(file),
+          )
+          importedAssets.forEach((file) =>
+            chunk.viteMetadata!.importedAssets.add(file),
+          )
+          return false
+        }
+        return true
+      })
+      chunk.code = replaceEmptyChunk(chunk.code)
+    }
+  }
+}
```

Comment:
I think we can put this code back inside the Vite plugin and test it through e2e instead. It's more hands-on to be able to safely unit test it.

We can have the `getEmptyChunkReplacer` only extracted and unit tested for now.